### PR TITLE
Add env vars for dc, region

### LINKED ISSUE #2183: [rfc] add NOMAD_DC and NOMAD_REGION to ENV
Hi,

Would it make sense (and be possible) to have `NOMAD_DC` and `NOMAD_REGION` available in the env for an allocation.

### PR DESCRIPTION
**Pull request by qkate**
Closes #2183

- adds env vars for datacenter and region to job env
- makes them both available for interpolation from within jobs

### CODE REVIEW

**Comment by jippi** [comment]
Nice @qkate !

File: nomad/structs/structs.go [hunk 1]
```diff
@@ -708,6 +708,9 @@ type Node struct {
 	// Datacenter for this node
 	Datacenter string
 
+	// Region for this node
```
Comment (dadgar) [inline nomad/structs/structs.go]: Hey lets actually remove this. It is better to keep this normalized as we will not have a field string stored in memory for every node that is know by virtue of the region you are talking to.

File: client/driver/driver.go [hunk 2]
```diff
@@ -278,6 +278,8 @@ func GetTaskEnv(taskDir *allocdir.TaskDir, node *structs.Node,
 	env := env.NewTaskEnvironment(node).
 		SetTaskMeta(alloc.Job.CombinedTaskMeta(alloc.TaskGroup, task.Name)).
 		SetJobName(alloc.Job.Name).
+		SetDatacenterName(node.Datacenter).
+		SetRegionName(node.Region).
```
Comment (dadgar) [inline client/driver/driver.go]: conf.Region

File: website/source/docs/runtime/environment.html.md [hunk 3]
```diff
@@ -61,6 +61,14 @@ environment variables.
     <td>The job's name</td>
   </tr>
   <tr>
+    <td>`NOMAD_DC`</td>
+    <td>The datacenter in which the job is running.</td>
```
Comment (sethvargo) [inline website/source/docs/runtime/environment.html.md]: Pedantic, but these aren't sentences, so we should not have a period at the end 😄

**Comment by qkate** [comment]
@sethvargo -- roger that, I removed the periods!

File: client/driver/env/env.go [hunk 4]
```diff
@@ -227,6 +242,7 @@ func (t *TaskEnvironment) Build() *TaskEnvironment {
 		// Set up the node values.
 		t.NodeValues[nodeIdKey] = t.Node.ID
 		t.NodeValues[nodeDcKey] = t.Node.Datacenter
+		t.NodeValues[nodeRegionKey] = t.Node.Region
```
Comment (qkate) [inline client/driver/env/env.go]: @dadgar I'm guessing I should remove this line as well?

File: command/agent/agent.go [hunk 5]
```diff
@@ -275,6 +275,7 @@ func (a *Agent) clientConfig() (*clientconfig.Config, error) {
 	// Setup the node
 	conf.Node = new(structs.Node)
 	conf.Node.Datacenter = a.config.Datacenter
+	conf.Node.Region = a.config.Region
```
Comment (qkate) [inline command/agent/agent.go]: @dadgar And should I remove this too?

Comment (dadgar) [inline command/agent/agent.go]: Yep

File: client/driver/env/env.go (hunk 4 above)
Comment (dadgar) [inline client/driver/env/env.go]: I would change it to `t.NodeValues[nodeRegionKey] = t.Region

File: website/source/docs/runtime/environment.html.md [hunk 6]
```diff
@@ -61,6 +61,14 @@ environment variables.
     <td>The job's name</td>
   </tr>
   <tr>
+    <td>`NOMAD_DC`</td>
+    <td>The datacenter in which the job is running</td>
```
Comment (dadgar) [inline website/source/docs/runtime/environment.html.md]: job -> allocation

File: website/source/docs/runtime/environment.html.md [hunk 7]
```diff
@@ -61,6 +61,14 @@ environment variables.
...
+  </tr>
+  <tr>
+    <td>`NOMAD_REGION`</td>
+    <td>The region in which the job is running</td>
```
Comment (dadgar) [inline website/source/docs/runtime/environment.html.md]: Same here

**Comment by wilkmaia** [comment]
https://travis-ci.org/hashicorp/nomad/builds/228530053#L2135

**Comment by dadgar** [comment]
@wilkmaia that is a somewhat flaky test so given all else passed and they are unrelated I think it is a red herring! But thanks for noticing!

**Pull request merged**